### PR TITLE
[Infra] Remove CoreDiagnostics library and references

### LINKED ISSUE #10192: [Firebase 10] Remove CoreDiagnostics SDK
### Description

The SDK is no longer needed and can be removed.

### Reproducing the issue

_No response_

### Firebase SDK Version

9.5.0

### Xcode Version

13.3.1

### Installation Method

Swift Package Manager

### Firebase Product(s)

All

### Targeted Platforms

iOS, tvOS, watchOS, macOS, macCatalyst

### Relevant Log Output

_No response_

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Replace this line with the contents of your Package.resolved.

```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

### PR DESCRIPTION
**Pull request by ncooke3**
### Context
- Remove CoreDiagnostics as it is no longer needed.
- There are no references when grepping the repo for  'CoreDiagnostics'.
   ```shell
   ➜  firebase-ios-sdk git:(nc/remove-cd) 
   ‣ git grep 'CoreDiagnostics'  
   ➜  firebase-ios-sdk git:(nc/remove-cd) 
   ```

#no-changelog

Fixes #10192

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -15,3 +15,5 @@ source 'https://rubygems.org'
 gem 'cocoapods', '1.11.3'
 gem 'cocoapods-generate', '2.0.1'
 gem 'danger', '8.4.5'
+
+# Run all CI
```
Comment (ncooke3) [inline Gemfile:19]: Revert before merging.

File: FirebaseCore/Sources/FIRApp.m [hunk 2]
```diff
@@ -46,26 +45,6 @@
 
 #import <objc/runtime.h>
 
-// The kFIRService strings are only here while transitioning CoreDiagnostics from the Analytics
-// pod to a Core dependency. These symbols are not used and should be deleted after the transition.
```
Comment (ncooke3) [inline FirebaseCore/Sources/FIRApp.m:50]: git grep -E 'kGGLService|kFIRService' outputs nothing so these weren't used anywhere.

**Comment by ncooke3** [comment]
The only failure is the `functions / quickstart` job and that is because of FirebaseUI not being compatible with Firebase 10.

File: FirebaseCore/Sources/FIRApp.m (hunk 2 above)
Comment (paulb777) [reply FirebaseCore/Sources/FIRApp.m:50]: This might need a global tap to confirm

**Review by paulb777 — APPROVED** [review]
Nice work!

File: FirebaseCore/Sources/FIRApp.m (hunk 2 above)
Comment (paulb777) [reply FirebaseCore/Sources/FIRApp.m:50]: nvm

**Pull request merged**